src/binary.ts

```diff
@@ -312,14 +222,11 @@ export class Binary {
   }
 
   /** @internal */
-  static fromExtendedJSON(
-    doc: { $type: string; $binary: string | { subType: string; base64: string } },
-    options?: EJSONOptions
-  ): Binary {
+  static fromExtendedJSON(doc: BinaryEJSON, options?: EJSONOptions): Binary {
     options = options || {};
     let data: Buffer | undefined;
     let type;
-    if (options.legacy && typeof doc.$binary === 'string') {
+    if (options.legacy && '$type' in doc && typeof doc.$binary === 'string') {
```

Comment:
I think this is still wrong, the original conditional is checking only for `doc.$binary`, the `doc.$type` is optional and defaults to 0 if its not present. Looks like we need a test for this if it's not triggering any breakage

Reply:
Aha! It was an issue with the interface, should be fixed now.